From 697d1648b796b6ba5c08dc01a86ed2e324690abf Mon Sep 17 00:00:00 2001
From: Akseli Lahtinen <akselmo@akselmo.dev>
Date: Wed, 29 Nov 2023 16:00:34 +0000
Subject: [PATCH] Set style to nullptr before painting listview

Resolves any possible dangling pointers
if application style is changed after
toggling the selection mode

BUG:477288
---
 src/views/dolphinview.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/views/dolphinview.cpp b/src/views/dolphinview.cpp
index 3ada359a55..4e74245c12 100644
--- a/src/views/dolphinview.cpp
+++ b/src/views/dolphinview.cpp
@@ -308,8 +308,8 @@ void DolphinView::setSelectionModeEnabled(const bool enabled)
         m_view->setStyle(m_proxyStyle.get());
         m_view->setEnabledSelectionToggles(DolphinItemListView::SelectionTogglesEnabled::False);
     } else {
-        setStyle(QApplication::style());
-        m_view->setStyle(QApplication::style());
+        setStyle(nullptr);
+        m_view->setStyle(nullptr);
         m_view->setEnabledSelectionToggles(DolphinItemListView::SelectionTogglesEnabled::FollowSetting);
     }
     m_container->controller()->setSelectionModeEnabled(enabled);
-- 
GitLab

